src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -327,7 +343,6 @@ static CloseableHttpClient buildHttpClient(
 
       HttpClientBuilder httpClientBuilder =
           HttpClientBuilder.create()
-              .setDefaultRequestConfig(DefaultRequestConfig)
```

Comment:
So this was the source of the problem? Like we set the config before we check whether proxy is used?

Reply:
Yes- this was part of it. The other part is that HTTP protocol is used by default, and HTTPS was not getting specified at all.